presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1421,6 +1429,23 @@ void PrestoServer::reportNodeStatus(proxygen::ResponseHandler* downstream) {
   http::sendOkResponse(downstream, json(fetchNodeStatus()));
 }
 
+void PrestoServer::handleGracefulShutdown(
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body,
+    proxygen::ResponseHandler* downstream) {
+  if (body.size() == 1 &&
+      folly::trimWhitespace(body[0]->moveToFbString()) == "\"SHUTTING_DOWN\"") {
+    LOG(INFO) << "Shutdown requested";
+    if (nodeState() == NodeState::kActive) {
+      std::thread([this]() { this->stop(); }).detach();
```

Comment:
Why do we invoke stop in a separate thread? Please add a comment.

Reply:
Added comment.